accelerometer/processing.py

```diff
@@ -0,0 +1,356 @@
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+from scipy.signal import butter, sosfiltfilt, welch
+from scipy.stats import entropy, median_abs_deviation
+from scipy.ndimage import median_filter
+from tqdm.auto import tqdm
+
+
+GRAVITY_CUTOFF_HZ = 0.5
+NOISE_CUTOFF_HZ = 20
+MAXVAL, MINVAL = 6, -6  # Note: GENEActiv has a range of +/-6g whereas Axivity and Actigraph +/-8 or more
+
+
+class Processing():
+    def __init__(self, **kwargs):
+        self.sampleRate = kwargs['sampleRate']
+        self.epochPeriod = kwargs['epochPeriod']
+        self.epochFile = kwargs['epochFile']
+        self.timeZone = kwargs['timeZone']
+        self.extractor = None
+        if kwargs.get('extractFeatures', True):
+            self.extractor = FeatureExtractor(
+                sampleRate=self.sampleRate,
+                epochPeriod=self.epochPeriod
+            )
+
+
+    def run(self, npyFile):
+        data = loadNpyToFrame(npyFile)
+
+        #TODO quality control
+
+        data = Processing.regularizeSampleRate(data, self.sampleRate)
+        data = Processing.filterNoise(data, self.sampleRate)
+
+        #TODO
+        # Used to be done in device.py but would make sense to do it here.
+        # data = calibrateGravity(...)
+
+        #TODO
+        # Currently done in summariseEpoch but would make more sense to do it
+        # here. Also the approach is slightly different to the paper.
+        # data = detectNonwear(...)
```

Comment:
We would also like this function to output the start/end times of each non-wear episode (to support later imputation)